packages/block/block.ts

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import {
+  childNodes$,
+  cloneNode$,
+  createEventListener,
+  insertBefore$,
+  insertText,
+  innerHTML$,
+  remove$ as removeElement$,
+  setAttribute,
+  setText,
+  mapGet$,
+  mapHas$,
+  mapSet$,
+  setStyleAttribute,
+  setSvgAttribute,
+} from './dom';
+import { renderToTemplate } from './template';
+import { AbstractBlock, Hole } from './types';
+import type { Edit, EditChild, Props, VElement } from './types';
+
+export const createBlock = (fn: (props?: Props) => VElement) => {
+  const cache = new Map<string, Hole>();
```

Comment:
@aidenybai Most of the time, the `createBlock` will be called at the top level of a module:

```tsx
const row = createBlock(({ value }) => {
  return <li>{value}</li>;
});
```

Thus the block will never be GC-ed, which also means the `cache` will never be released as well.